disassemblers/ofrak_angr/ofrak_angr/components/angr_analyzer.py

```diff
@@ -69,3 +73,38 @@ def _create_dependencies(
         Step 2. Modification
         Step 3. Packing
         """
+
+
+class AngrCodeRegionModifier(Modifier):
+    id = b"AngrCodeRegionModifier"
+    targets = (CodeRegion,)
+
+    async def modify(self, resource: Resource, config=None):
+        code_region = await resource.view_as(CodeRegion)
+
+        root_resource = await resource.get_only_ancestor(
+            ResourceFilter(tags=[AngrAnalysisResource], include_self=True)
+        )
+
+        fixup_address = False
+
+        # We only want to adjust the address of a CodeRegion if the original binary is position-independent.
+        # Implement PIE-detection for other file types as necessary.
+        if root_resource.has_tag(Elf):
+            elf_header = await root_resource.get_only_descendant_as_view(
+                ElfHeader, r_filter=ResourceFilter(tags=[ElfHeader])
+            )
+
+            if elf_header is not None and elf_header.e_type == ElfType.ET_DYN.value:
+                fixup_address = True
+        else:
+            LOGGER.warning(
+                f"Have not implemented PIE-detection for {root_resource}. The address of {code_region} will likely be incorrect."
```

Comment:
I'm not sure if we should say "likely" incorrect here. I suppose most of the random ELFs one might come across are PIE, but is the same true of other files.

The uncertainty here again points to why I think it would be ideal to check specifically if the memory map in Angr and the ELF headers are different. While CodeRegions are only executable regions, we could get the lowest `ProgramSection` and compare that to the start address in Angr (`obj.min_addr` right?). This method would 1) not only apply to ELFs, but all programs, 2) not assume that only PIE can cause a memory map mismatch, 3) check _directly_ for the memory map mismatch. Is there a reason that wouldn't work?

Reply:
Yeah I think that would work

Reply:
I tried this and it actually does not quite work because of sections not loaded into the binary (such as .comment or .strtab). These sections have a virtual address of 0 and hence ruin the minimum calculation. As I believe there's no way in OFRAK to distinguish between sections we would and would not want to count in this calculation, I think the PIE-detecting route is our best option.

Reply:
Until we know more about the Angr memory mapping APIs, we'll go with this. Supporting PIE ELF files is better than not supporting PIE files.